flink-clients/src/main/java/org/apache/flink/client/CliFrontend.java

```diff
@@ -517,67 +521,83 @@ protected int list(String[] args) {
 				return 1;
 			}
 
-			Iterable<ExecutionGraph> jobs = AkkaUtils.<RunningJobs>ask(jobManager,
-					RequestRunningJobs$.MODULE$, getAkkaTimeout()).asJavaIterable();
+			final Future<Object> response = Patterns.ask(jobManager,
+					JobManagerMessages.getRequestRunningJobs(), new Timeout(getAkkaTimeout()));
 
-			ArrayList<ExecutionGraph> runningJobs = null;
-			ArrayList<ExecutionGraph> scheduledJobs = null;
-			if (running) {
-				runningJobs = new ArrayList<ExecutionGraph>();
-			}
-			if (scheduled) {
-				scheduledJobs = new ArrayList<ExecutionGraph>();
+			Object result = null;
+
+			try{
+				result = Await.result(response, getAkkaTimeout());
+			} catch (Exception exception) {
+				throw new IOException("Could not retrieve running jobs from job manager.",
+						exception);
 			}
-			
-			for (ExecutionGraph rj : jobs) {
-				
-				if (running && rj.getState().equals(JobStatus.RUNNING)) {
-					runningJobs.add(rj);
+
+			if(!(result instanceof RunningJobs)){
```

Comment:
More like nit style change. Most code in Flink add extra space between 'if' and the '(' and also a space before '{'.